### DISCUSSION TITLE
Wrap poetry

### DISCUSSION THREAD

**Discussion by luckydonald** [nat-n/poethepoet#210] [Q&A]
So the initial idea of poe was to extend poetry to have the `poetry run` command.

(However poetry run does something else.)

Now this tool tries to give this missing run command, but introduces a new command instead of extending poetry.

- `poetry add somepackage`
- `poe tests`

What I propose is a mode (e.g. configured through the toml) which wraps poetry to have it a single executable to remember.

- `poe add somepackage`
- `poe run tests`

**Comment by nat-n** [comment]
Hi @luckydonald, thanks for the idea. I guess what you mean is that the `poe` CLI would by default behave like an alias for the `poetry` CLI, but if you define poe tasks then they're added to the tasks available in poetry?

This sounds almost like what you get if you install the [poethepoet poetry plugin](https://poethepoet.natn.io/poetry_plugin.html) and configure `tool.poe.poetry_command = ""`. In this configuration poe tasks are added directly to the `poetry` CLI. Have you tried that?

**Comment by luckydonald** [comment]
Ah, I tried installing the poetry as plugin and that didn't quite work, I skipped that part because it was so confusing, and seemed like an extra, and I didn't understand how it would work like you just explained. Though I was searching for that.
Especially as I came from a stack overflow page where you advertised your plugin as a solution.

**Reply by nat-n** [reply]
If you installed poetry via the installer script (recommended) then installing the plugin with the following should just work.

```sh
poetry self add 'poethepoet[poetry_plugin]'
```

I'm be curious to know details if you encounter any issue with this.

**Comment by luckydonald** [comment]
> Whatever tool.poe.poetry_command is set to must not already exist as a poetry command!

What would happen then? That kinda scared me off, too.

**Reply by nat-n** [reply]
If there is a collision in task names between poe and poetry then the plugin will raise an exception with a helpful error message. This is a fundamental limitation with the idea of adding tasks to poetry, and I don't think there's a better solution. It's generally easy to avoid.
